ShapeEnv.create_symbolic_sizes_strides_storage_offset

ezyang opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #90295
* #90286
* #90202
* #90186
* #90185
* #90179
* #90009
* __->__ #89962

Instead of having storage offset hang out on its own, allocate
all of these symbols all in one go.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

ezyang committed:
ShapeEnv.create_symbolic_sizes_strides_storage_offset

Instead of having storage offset hang out on its own, allocate
all of these symbols all in one go.

Signed-off-by: Edward Z. Yang <ezyang@fb.com>

[ghstack-poisoned]

albanD reviewed (approved):
Sounds good!
I guess this is towards being able to specialize differently storage offset from the other symints?

ezyang commented:
> I guess this is towards being able to specialize differently storage offset from the other symints?

Yeah, it makes it easier to associate storage offset with the tensor in question

ezyang committed:
Update on "ShapeEnv.create_symbolic_sizes_strides_storage_offset"


Instead of having storage offset hang out on its own, allocate
all of these symbols all in one go.

Signed-off-by: Edward Z. Yang <ezyangfb.com>

[ghstack-poisoned]

ezyang commented:
@pytorchbot merge -f "merge bot is WRONG"
